tests/components/tagging.directive.spec.js

```diff
@@ -0,0 +1,192 @@
+describe('app.components.taggingWidget directive', function() {
+  var $scope;
+  var $compile;
+  var element;
+  var collectionsApiSpy;
+  var isoScope;
+  var options = {
+    expand: 'resources',
+  };
+
+  var attributes = ['categorization', 'category.id', 'category.single_value'];
+  var options2 = {
+    expand: 'resources',
+    attributes: attributes,
+  };
+
+  beforeEach(function () {
+    module('app.components', 'app.services', 'app.config', 'app.states', 'gettext');
+    bard.inject('$state', 'Session', '$httpBackend', 'CollectionsApi', '$timeout');
+  });
+
+  beforeEach(inject(function (_$compile_, _$rootScope_, _$document_) {
+    $compile = _$compile_;
+    $scope = _$rootScope_;
+    $document = _$document_;
+  }));
+
+  var compileHTML = function (markup, scope) {
+    element = angular.element(markup);
+    var el = $compile(element)(scope);
+    scope.$digest();
+    isoScope = el.isolateScope();
+    isoScope.vm.loadAllCategories();  //this chained promise isn't being called -call manually
+  };
+
+  beforeEach(function () {
+    Session.create({
+      auth_token: 'b10ee568ac7b5d4efbc09a6b62cb99b8',
+    });
+    $httpBackend.whenGET('').respond(200);
+
+    var mockDir = 'tests/mock/blueprint-details/';
+    $scope.tagsOfBlueprint = readJSON(mockDir + 'tags-of-blueprint.json');
+    $scope.allTags = readJSON(mockDir + 'all-tags.json');
+    $scope.tagCategories = readJSON(mockDir + 'tag-categories.json');
+
+    collectionsApiSpy = sinon.stub(CollectionsApi, 'query');
+    collectionsApiSpy.withArgs('categories', options).returns(Promise.resolve($scope.tagCategories));
+    collectionsApiSpy.withArgs('tags', options2).returns(Promise.resolve($scope.allTags));
+
```

Comment:
@jeff-phillips-18 Figured out how to use sinon.stub.withArgs to return data from promises :-)